plugin/completion.py

```diff
@@ -171,29 +171,15 @@ def initialize(self) -> None:
             # This is to make ST match with labels that have a weird prefix like a space character.
             self.view.settings().set("auto_complete_preserve_order", "none")
 
-    def _view_language(self, config_name: str) -> Optional[str]:
-        languages = self.view.settings().get('lsp_language')
-        return languages.get(config_name) if languages else None
-
     def register_trigger_chars(self, session: Session, trigger_chars: List[str]) -> None:
-        completion_triggers = self.view.settings().get('auto_complete_triggers', []) or []  # type: List[Dict[str, str]]
-        view_language = self._view_language(session.config.name)
-        if view_language:
-            for language in session.config.languages:
-                if language.id == view_language:
-                    for scope in language.scopes:
-                        # debug("registering", trigger_chars, "for", scope)
-                        scope_trigger = next(
-                            (trigger for trigger in completion_triggers if trigger.get('selector', None) == scope),
-                            None
-                        )
-                        if not scope_trigger:  # do not override user's trigger settings.
-                            completion_triggers.append({
-                                'characters': "".join(trigger_chars),
-                                'selector': scope
-                            })
-
-            self.view.settings().set('auto_complete_triggers', completion_triggers)
+        completion_triggers = self.view.settings().get('auto_complete_triggers') or []  # type: List[Dict[str, str]]
+
+        completion_triggers.append({
+            'characters': "".join(trigger_chars),
+            'selector': "- comment - punctuation.definition.string.end"
+        })
```

Comment:
This is starting to look good :)
The `auto_complete_triggers` list is stored in the .sublime-workspace file in case the user is using a .sublime-project, and otherwise in the .sublime-session file in $DATA/Local/session.sublime-session. Did you verify whether these changes are not adding an infinite amount of entries to the `auto_complete_triggers` list every time ST is restarted? The old code made sure that that didn't happen.

On that note, it's maybe good to rethink this too. ST ignores unknown keys in an auto_complete_trigger dict, so what we can do is add a `server` key like this:

```python
{
    'characters': "".join(trigger_chars),
    'selector': '- comment - punctuation.definition.string.end'
    'server': session.config.name
}
```

We can then remove our enrties in the `__del__` method of the listener:

```python

    def __del__(self) -> None:
        triggers = self.view.settings().get("auto_complete_triggers")
        triggers = [trigger for trigger in triggers if 'server' not in trigger]
        # put triggers back into auto_complete_triggers ...
```

Reply:
> adding an infinite amount of entries to the auto_complete_triggers list 

Short anwser:
To answer your question, it won't append the list to infinity,
so we don't need a `server` key in the dict.

Long answer:
That was one of the first things I checked. 
Each time you open for example a index.js file.
And you did not trigger the AC.
if you run in the sublime console this:
```
>>> view.settings().get('auto_complete_triggers')
[{'characters': '<', 'selector': 'text.html, text.xml'}, {'selector': 'punctuation.accessor'}]
```

If you trigger the AC, the output will be different
```
>>> view.settings().get('auto_complete_triggers')
[{'characters': '<', 'selector': 'text.html, text.xml'}, {'selector': 'punctuation.accessor'}, {'characters': '."\'/@<', 'selector': '- comment - punctuation.definition.string.end'}]
```
if you trigger the AC again you will get the same output as before.

If you open a new file `main.js`.
And you did not trigger the AC.
if you run in the sublime console this:
```
>>> view.settings().get('auto_complete_triggers')
[{'characters': '<', 'selector': 'text.html, text.xml'}, {'selector': 'punctuation.accessor'}]
```

If you trigger the AC, the output will be different
```
>>> view.settings().get('auto_complete_triggers')
[{'characters': '<', 'selector': 'text.html, text.xml'}, {'selector': 'punctuation.accessor'}, {'characters': '."\'/@<', 'selector': '- comment - punctuation.definition.string.end'}]
```